include/llvm/ADT/ilist.h

```diff
@@ -420,7 +420,21 @@ class iplist : public Traits {
       Head = New;
     this->setPrev(CurNode, New);
 
-    this->addNodeToList(New);  // Notify traits that we added a node...
+    // HLSL Change Begin: Undo insertion if exception
+    try {
+      this->addNodeToList(New);  // Notify traits that we added a node...
```

Comment:
shouldn't addNotToList deal with this?

Reply:
`addNodeToList` is more like calling a listener. The current method is what hooks the node in the linked list.